packages/dhis2/src/Adaptor.js

```diff
@@ -634,24 +605,54 @@ export function upsert(
       // NOTE: that these parameters are all expanded by the `get`, `create`, and
       // `update` functions used inside this composed "upsert" function.
       console.log(`Preparing upsert via 'get' then 'create' OR 'update'...`);
-      promise = get(
-        resourceType,
-        query,
-        options
-      )(state).then(resp => {
-        const resources = resp.data[resourceType];
+      promise = request(configuration, {
```

Comment:
Can we refactor this, and other similar structures, to use async/await? While we're here?